Neos.Neos/Classes/Domain/NodeLabel/NodeLabelRenderer.php

```diff
@@ -2,24 +2,40 @@
 
 declare(strict_types=1);
 
-namespace Neos\ContentRepositoryRegistry\Factory\NodeTypeManager;
+namespace Neos\Neos\Domain\NodeLabel;
 
-use Neos\ContentRepository\Core\NodeType\NodeLabelGeneratorFactoryInterface;
-use Neos\ContentRepository\Core\NodeType\NodeLabelGeneratorInterface;
 use Neos\ContentRepository\Core\NodeType\NodeType;
-use Neos\ContentRepositoryRegistry\NodeLabel\ExpressionBasedNodeLabelGenerator;
+use Neos\ContentRepository\Core\Projection\ContentGraph\Node;
+use Neos\ContentRepositoryRegistry\ContentRepositoryRegistry;
 use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Neos\Neos\Fusion\Helper\NodeHelper;
+use Neos\Neos\Utility\NodeTypeWithFallbackProvider;
 
-class ObjectManagerBasedNodeLabelGeneratorFactory implements NodeLabelGeneratorFactoryInterface
+/**
+ * @api For PHP, in Fusion one can use ${Neos.Node.renderLabel(node)} {@see NodeHelper::renderLabel()}
+ */
+final readonly class NodeLabelRenderer
```

Comment:
todo what if someone wants to replace the whole label renderer? And make singleton